src/main/java/dk/dbc/updateservice/update/UpdateServiceCore.java

```diff
@@ -205,6 +216,104 @@ public UpdateRecordResponseDTO updateRecord(UpdateServiceRequestDTO updateServic
         });
     }
 
+    public void updateRecord(RecordEntryDTO recordEntryDTO) throws UpdateException {
+        try {
+            final RecordId recordId = RecordDTOMapper.getRecordId(recordEntryDTO);
+            final String bibliographicRecordId = recordEntryDTO.getRecordId().getBibliographicRecordId();
+            final int agencyId = recordEntryDTO.getRecordId().getAgencyId();
+            final Record rawRecord = rawRepo.fetchRecord(bibliographicRecordId, agencyId);
+            final List<RecordId> relationParents;
+
+            RecordDTOMapper.toRecord(recordEntryDTO, rawRecord);
+
+            // Set relations for active record. Deleted record don't have any relations
+            if (!rawRecord.isDeleted()) {
+                if (MarcXChangeMimeType.ENRICHMENT.equals(rawRecord.getMimeType())) {
+                    relationParents = Collections.singletonList(getCommonRecord(bibliographicRecordId, agencyId));
+                } else {
+                    // Look for referenced records
+                    final MarcRecord marcRecord = RecordDTOMapper.getMarcRecord(recordEntryDTO);
+                    relationParents = getParentRelations(marcRecord);
+                }
+            } else {
+                relationParents = Collections.emptyList();
+            }
+
+            final LibraryGroup libraryGroup = vipCoreService.getLibraryGroup(Integer.toString(agencyId));
+            final String providerId = EnqueueRecordAction.getProvider(settings, libraryGroup);
+            rawRepo.saveRecord(rawRecord);
+            rawRepo.removeLinks(recordId);
+            for (RecordId relation : relationParents) {
+                makeSureParentRecordExists(relation.getBibliographicRecordId(), relation.getAgencyId());
+                rawRepo.linkRecordAppend(recordId, relation);
+            }
+
+            rawRepo.enqueue(recordId, providerId, true, false, 1001);
```

Comment:
magic number 1001

Reply:
Enig, det er meget magisk. Spørgsmålet er hvilken prioritet posterne fra det endpoint skal ind med. Man kunne f.eks. vælge en magisk konstant, eller konfigurerer det med en environment variable. Men jeg har ændret det til default, dvs 500. Hvis det giver problemer må vi fikse det.

Reply:
Njah, 1001 er for så vidt godt nok, men det ville være mere forklarende at have en konstant med den værdi og at den så f.eks. hed LOW_PRIORITY.